.github/workflows/ci.yaml

```diff
@@ -50,21 +50,46 @@ jobs:
   test_agent_image:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - name: Build image
-      run: docker build . --file mirrord-agent/Dockerfile
+      - uses: actions/checkout@v2
+      - name: Build image
+        run: docker build . --file mirrord-agent/Dockerfile
 
-  
   test_mirrord_layer:
     strategy:
       matrix:
         os: [ubuntu-latest, ubuntu-18.04, macos-12, macos-11, macos-10.15]
     runs-on: ${{ matrix.os }}
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions-rs/toolchain@v1
-      with:
-        profile: minimal
-        toolchain: nightly
-    # For now, just evrify it compiles.
-    - run: cargo +nightly build --manifest-path=mirrord-layer/Cargo.toml
\ No newline at end of file
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+      # For now, just evrify it compiles.
+      - run: cargo +nightly build --manifest-path=mirrord-layer/Cargo.toml
+
+  e2e:
+    runs-on: ubuntu-latest
```

Comment:
Can we try and do a platform matrix here? Lets start with mac and ubuntu.

Reply:
Save this for later though - let's get this PR merged first.

Reply:
yes, i'll add macos

Reply:
added to my todo list, will open an issue later